### PR TITLE
Improve Design and Browserfunctionality

### PR DESCRIPTION
**Pull request by ivy-edp**
Added doubleclick-insert to cms- & datatype-browser. Should I also add it to the attribute-browser?
Improved table- and helpertextheight for the browsers
Adjusted the paddingleft for Treetables/removed blue cirle
Made simpleName in Datatype-Browser bold
Color-Changes: 
- Better contrast (as Sprint-Review suggested)
- New Hover and Selectstate-color "blue" for Input/ScriptInput/Combobox/checkbox/radio/table

### CODE REVIEW

File: packages/editor/src/components/widgets/table/cell/ExpandableCell.tsx [hunk 1]
```diff
@@ -39,12 +40,18 @@ export function ExpandableCell<TData>({ cell, isLoaded, loadChildren, isUnknown,
...
+          <span className='row-expand-label'>
+            <b>{cell.getValue() as string}</b>
+          </span>
+          <span className='row-expand-label'>- {helperInfo}</span>
         </>
-      )}{' '}
-      <span className='row-expand-label'>{cell.getValue() as string}</span>
+      ) : (
+        <span className='row-expand-label'>{cell.getValue() as string}</span>
+      )}
```
Comment (ivy-lli) [inline packages/editor/src/components/widgets/table/cell/ExpandableCell.tsx:54]: instead of this big conditional stuff move the additional info more to the place where it is needed, so you can reduce duplication!
And e.g. use css to make the difference between normal text and addional info

```jsx
<span className='row-expand-label'>{cell.getValue() as string}</span>
{helperInfo ?? <span className='row-expand-label-info'>- {helperInfo}</span>}
```

File: packages/editor/src/components/widgets/table/cell/ExpandableCell.tsx [hunk 2]
```diff
@@ -11,9 +11,10 @@ type ExpandableCellProps<TData> = {
   isUnknown?: boolean;
   icon?: IvyIcons;
   title?: string;
+  helperInfo?: string;
 };
 
-export function ExpandableCell<TData>({ cell, isLoaded, loadChildren, isUnknown, icon, title }: ExpandableCellProps<TData>) {
+export function ExpandableCell<TData>({ cell, isLoaded, loadChildren, isUnknown, icon, title, helperInfo }: ExpandableCellProps<TData>) {
```
Comment (ivy-lli) [inline packages/editor/src/components/widgets/table/cell/ExpandableCell.tsx:17]: don't really like the name helperinfo. additionalInfo or something like that is more suitable I think
And helperinfo is really near to the helper text in the browser

File: packages/editor/src/components/widgets/table/row/SelectRow.css [hunk 3]
```diff
@@ -2,9 +2,16 @@
...
 
 .selectable-row:where([data-state='selected']) {
-  background-color: var(--N100);
+  background-color: var(--A400);
+  color: var(--background);
+}
+
+.selectable-row:where([data-state='selected']:last-child) {
+  background-color: var(--A400);
+  color: var(--background);
```
Comment (ivy-lli) [inline packages/editor/src/components/widgets/table/row/SelectRow.css:15]: duplicated from above -> not needed...

File: packages/editor/src/components/widgets/table/row/SelectRow.css [hunk 4]
```diff
@@ -2,9 +2,16 @@
...
+  border-radius: var(--border-radius) !important;
```
Comment (ivy-lli) [inline packages/editor/src/components/widgets/table/row/SelectRow.css:16]: why is this needed???

File: packages/editor/src/components/widgets/table/row/SelectRow.tsx [hunk 5]
```diff
@@ -6,13 +6,24 @@ type SelectRowProps<TData> = {
...
+      if (event.detail === 1) {
+        if (isNotSelectable) {
+          undefined;
+        } else {
+          row.getToggleSelectedHandler()(event);
+        }
+      } else if (onDoubleClick) {
+        onDoubleClick();
+      }
+    }}
```
Comment (ivy-lli) [inline packages/editor/src/components/widgets/table/row/SelectRow.tsx]: instead of undefined, turn the condition:
```ts
if (event.detail === 1) {
  if (!isNotSelectable) {
    row.getToggleSelectedHandler()(event);
  }
} else if (onDoubleClick) {
  onDoubleClick();
}
```

File: packages/editor/src/components/widgets/table/table/Table.css [hunk 6]
```diff
@@ -19,3 +18,15 @@
 .table tr :where(td) {
   border-top: var(--seperator-border);
 }
+
+.table tr:last-child :where(td:first-child) {
```
Comment (ivy-lli) [inline packages/editor/src/components/widgets/table/table/Table.css:22]: where is only needed if you use more than one selector in it
reduce complication -> `.table tr:last-child td:first-child {`

File: packages/editor/src/components/widgets/table/table/Table.css [hunk 7]
```diff
@@ -19,3 +18,15 @@
...
+  border-bottom-left-radius: var(--border-radius);
+}
+
+.table tr:last-child :where(td:last-child) {
```
Comment (ivy-lli) [inline packages/editor/src/components/widgets/table/table/Table.css:26]: same here

File: packages/editor/src/components/browser/Browser.tsx [hunk 8]
```diff
@@ -17,27 +17,34 @@ type BrowserProps = UseBrowserReturnValue & {
...
+    onOpenChange(false);
+    acceptBrowser();
+  };
+
   const attrBrowser = useAttributeBrowser(location);
-  const cmsBrowser = useCmsBrowser(cmsOptions);
+  const cmsBrowser = useCmsBrowser(onBrowserRowDoubleClick, cmsOptions);
   const funcBrowser = useFuncBrowser();
-  const dataTypeBrowser = useDataTypeBrowser();
+  const dataTypeBrowser = useDataTypeBrowser(onBrowserRowDoubleClick);
```
Comment (ivy-lli) [inline packages/editor/src/components/browser/Browser.tsx:40]: I know that you need to add this to the browsers, but honestly I think I would expect this behaviour on every browser!
Because otherwise, no one will understand why it will work on this browser but not on the other one...

**Review by ivy-lli — APPROVED** [review]
Looks good!
Some small comments below

And in general, I would only use optional attributes only if you have a use case. So first work with required params and if you have the use case, make it optional

File: packages/editor/src/components/widgets/table/row/SelectRow.css (hunk 4 above)
Comment (ivy-edp) [reply packages/editor/src/components/widgets/table/row/SelectRow.css:16]: You are correct; it is no longer necessary. I encountered an issue where the border-radius from the table.css was not being applied. In an attempt to address this, I applied the following CSS. However, I have removed it, and everything is still works fine.

File: packages/editor/src/components/browser/Browser.tsx (hunk 8 above)
Comment (ivy-edp) [reply packages/editor/src/components/browser/Browser.tsx:40]: Yes, I agree with you. We can certainly brainstorm on how to solve this more effectively 👍

File: packages/editor/src/components/widgets/table/row/SelectRow.tsx [hunk 9]
```diff
@@ -6,13 +6,24 @@ type SelectRowProps<TData> = {
...
+    onClick={event => {
+      if (event.detail === 1) {
+        if (!isNotSelectable) {
+          row.getToggleSelectedHandler()(event);
+        }
+      } else if (onDoubleClick) {
+        if (!isNotSelectable) {
+          onDoubleClick();
+        }
+      }
```
Comment (ivy-lli) [inline packages/editor/src/components/widgets/table/row/SelectRow.tsx:25]: move isnotselectable to the start, instead of check it in each if block

Comment (ivy-edp) [reply packages/editor/src/components/widgets/table/row/SelectRow.tsx:25]: 👍
